addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -868,35 +868,19 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
      * @override
      */
     cleanForSave: function () {
-        const defs = [];
-
         this.$target[0].querySelectorAll('#editable_select').forEach(el => el.remove());
         const select = this._getSelect();
         if (select) {
             select.style.display = '';
         }
-        const dependencyEl = this._getDependencyEl();
-        if (dependencyEl) {
-            const fieldName = this.$target[0].querySelector('.s_website_form_input').name;
-            if (dependencyEl.closest('.s_website_form_field').dataset.visibilityDependency === fieldName) {
-                this._deleteConditionalVisibility(this.$target[0]);
-                const prom = new Promise((resolve, reject) => {
-                    const message = _t("You can't have 2 field visibility depending of each others. One of the 2 conditions has been removed.");
-                    Dialog.confirm(this, message, {
-                        cancel_callback: () => resolve(),
-                        buttons: [
-                            {
-                                text: _t("Ok"),
-                                classes: 'btn-primary',
-                                click: () => resolve(),
-                            },
-                        ],
-                    });
-                });
-                defs.push(prom);
-            }
+
+        // There might be no field to select in the conditional visibility list
+        // of field when enabling the visibility option. We need to clean the
+        // DOM and not let part of the attributes set.
+        const dependentFieldContainerEl = this.formEl.querySelectorAll(`.s_website_form_field_hidden_if:not([data-visibility-dependency])`);
+        for (const fieldContainerEl of dependentFieldContainerEl) {
+            this._deleteConditionalVisibility(fieldContainerEl);
```

Comment:
As discussed, maybe clean this directly during edition and with a dialog explaining to the user why setting a condition was not possible :+1:

Reply:
Changed, could you have a look?
As a reminder:
1. We now hide the field from the list if they create a direct or indirect dependency
2. If the list of field is empty, we warn the user and prevent the option to be enabled, note that such a case was already possible if you had only a one field form

About the indirect dependency, I just added it now, I forgot about it previously, but field A could depend of field B and field B could depend of field C, then we should prevent field A to be selected as dependency of field C.